src/marqo/core/inference/inference_models/hugging_face_model.py

```
@@ -0,0 +1,249 @@
+from marqo.tensor_search.models.private_models import ModelAuth
+
+
+class HuggingFaceModel(AbstractEmbeddingModel):
+    """The concrete class for all sentence transformers models loaded from Hugging Face."""
+
+    def __init__(self, model_properties: dict, device: str, model_auth: Optional[ModelAuth] = None):
+        super().__init__(model_properties, device, model_auth)
+
+        self.model_properties = self._build_model_properties(model_properties)
+
+        self._model = None
+        self._tokenizer = None
+        self._pooling_func = None
+
+    def _build_model_properties(self, model_properties: dict) -> HuggingFaceModelProperties:
+        """Convert the user input model_properties to HuggingFaceModelProperties."""
+        try:
+            return HuggingFaceModelProperties(**model_properties)
+        except ValidationError as e:
+            raise InvalidModelPropertiesError(f"Invalid model properties: {model_properties}. Original error {e}") \
+                from e
+
+    def _check_loaded_components(self):
+        if self._model is None:
+            raise InternalError("Model is not loaded!")
+        if self._tokenizer is None:
+            raise InternalError("Tokenizer is not loaded!")
+        if self._pooling_func is None:
+            raise InternalError("Pooling function is not loaded!")
+
+    def _load_necessary_components(self):
+        """Load the necessary components for the hf model.
+
+        Raises:
+            InvalidModelPropertiesError: If the model properties are invalid or incomplete.
+        """
+        if not (self.model_properties.name or self.model_properties.url or self.model_properties.model_location):
+            raise InvalidModelPropertiesError(
+                f"Invalid model properties for the 'hf' model. "
+                f"You do not have the necessary information to load the model. "
+                f"Check {marqo_docs.bring_your_own_model()} for more information."
+            )
```

nit. Is it better to do this check in _build_model_properties()?

---

make sense. I deleted this one, and raise an internal error in the next one since we should raise the error in the pydantic validation